File: e2e/playwright-python/interactions/pages/edit_task_form.py
from playwright.sync_api import Page, Locator
from data.task_status import TaskStatus
from data.task_priority import TaskPriority


class EditTaskForm:
    def __init__(self, page: Page):
        self._page = page
        self.title_input: Locator = page.get_by_test_id("edit-task-title-input")
        self.description_input: Locator = page.locator("#description")
        self.status_dropdown: Locator = page.get_by_test_id("status-dropdown")
        self.priority_dropdown: Locator = page.get_by_test_id("priority-dropdown")
        self.submit_button: Locator = page.get_by_test_id("save-button")

    def status_option(self, status: TaskStatus) -> Locator:
        return self._page.get_by_test_id(f"status-dropdown-option-{status.value}")

    def priority_option(self, priority: TaskPriority) -> Locator:
        return self._page.get_by_test_id(f"priority-dropdown-option-{priority.value}")

File: e2e/cypress-javascript/interactions/pages/EditTaskForm.js
class EditTaskForm {
  get titleInput() {
    return cy.get('[data-testid="edit-task-title-input"]');
  }

  get descriptionInput() {
    return cy.get('#description');
  }

  get statusDropdown() {
    return cy.get('[data-testid="status-dropdown"]');
  }

  get priorityDropdown() {
    return cy.get('[data-testid="priority-dropdown"]');
  }

  get submitButton() {
    return cy.get('[data-testid="save-button"]');
  }

  statusOption(status) {
    return cy.get(`[data-testid="status-dropdown-option-${status}"]`);
  }

  priorityOption(priority) {
    return cy.get(`[data-testid="priority-dropdown-option-${priority}"]`);
  }
}

module.exports = { EditTaskForm };
